src/ui.rs

```diff
@@ -161,7 +163,102 @@ fn draw_body<B: Backend>(
             }
         }))
         .graph_type(GraphType::Line)
-        .data(&historical_prices_data)];
+        .data(&historical_prices_data);
+    let mut historical_prices_datasets: Vec<_> = vec![];
+
+    let mut bb_prices_data = (vec![], vec![], vec![]);
+    let mut ema_prices_data = vec![];
+    let mut sma_prices_data = vec![];
+    if let Some(indicator) = ui_state.indicator {
+        let indicator_prices_data = stock.bars.iter().map(|bar| {
+            let mut data_item = DataItem::builder()
+                .open(bar.open)
+                .high(bar.high)
+                .low(bar.low)
+                .close(bar.close);
+            if let Some(volume) = bar.volume {
+                data_item = data_item.volume(volume as f64);
+            }
+            let data_item = data_item.build().unwrap();
+            (bar.timestamp_seconds() as f64, data_item)
+        });
+
+        match indicator {
+            Indicator::BollingerBands => {
+                let mut bb = indicators::BollingerBands::default();
+                let indicator_prices_data = indicator_prices_data.fold(
+                    (vec![], vec![], vec![]),
+                    |mut state, (timestamp, data_item)| {
+                        let bb_output = bb.next(&data_item);
+                        state.0.push((timestamp, bb_output.upper));
+                        state.1.push((timestamp, bb_output.average));
+                        state.2.push((timestamp, bb_output.lower));
+                        state
+                    },
+                );
+
+                bb_prices_data = indicator_prices_data.clone();
+            }
+            Indicator::ExponentialMovingAverage => {
+                let mut ema = indicators::ExponentialMovingAverage::default();
+                let indicator_prices_data: Vec<_> = indicator_prices_data
+                    .map(|(timestamp, data_item)| (timestamp, ema.next(&data_item)))
+                    .collect();
+
+                ema_prices_data = indicator_prices_data.clone();
+            }
+            Indicator::SimpleMovingAverage => {
+                // Yahoo Finance uses 50 as default
+                let mut sma = indicators::SimpleMovingAverage::new(50).unwrap();
+                let indicator_prices_data: Vec<_> = indicator_prices_data
+                    .map(|(timestamp, data_item)| (timestamp, sma.next(&data_item)))
+                    .collect();
+
+                sma_prices_data = indicator_prices_data.clone();
+            }
+        }
+    }
+    if Some(Indicator::BollingerBands) == ui_state.indicator {
```

Comment:
Prefer `match` over `if`-`else if`, as it enforces exhaustiveness. And it's also more readable.